include/can.h

```diff
@@ -0,0 +1,380 @@
+ * Define if the message has a standard (11bit) or extended (29bit)
+ * identifier
+ */
+enum can_ide {
+	CAN_STANDARD_IDENTIFIER,
+	CAN_EXTENDED_IDENTIFIER
+};
+
+/**
+ * @brief can_rtr enum
+ * Define if the message is a data or remote frame
+ */
+enum can_rtr {
+	CAN_DATAFRAME,
+	CAN_REMOTEREQUEST
+};
+
+/**
+ * @brief can_mode enum
+ * Defines the mode of the can controller
+ */
+enum can_mode {
+	/*Normal mode*/
+	CAN_NORMAL_MODE,
+	/*Controller is not allowed to send dominant bits*/
+	CAN_SILENT_MODE,
+	/*Controller is in loopback mode (receive own messages)*/
+	CAN_LOOPBACK_MODE,
+	/*Combination of loopback and silent*/
+	CAN_SILENT_LOOPBACK_MODE
+};
+
+/**
+ * @brief can message structure
+ *
+ * Used to pass can messages from userspace to the driver and
+ * from driver to userspace
+ *
+ */
+struct can_msg {
+	/** Indicates the identifier type (standard or extended)
+	 * use can_ide enum for assignment
+	 */
+	u32_t id_type : 1;
+	/** Set the message to a transmission request instead of data frame
+	 * use can_rtr enum for assignment
+	 */
+	u32_t rtr     : 1;
+	/** Message identifier*/
+	union {
+		u32_t std_id  : 11;
+		u32_t ext_id  : 29;
+	};
+	/** The length of the message (max. 8) in byte */
+	u8_t dlc;
+	/** The message data*/
+	union {
+		u8_t data[8];
+		u32_t data_32[2];
+	};
+} __packed;
+
+/**
+ * @brief can filter structure
+ *
+ * Used to pass can identifier filter information to the driver.
+ * rtr_mask and *_id_mask are used to mask bits of the rtr and id fields.
+ * If the mask bit is 0, the value of the corresponding bit in the id or rtr
+ * field don't care for the filter matching.
+ *
+ */
+struct can_filter {
+	/** Indicates the identifier type (standard or extended)
+	 * use can_ide enum for assignment
+	 */
+	u32_t id_type      : 1;
+	/** target state of the rtr bit */
+	u32_t rtr          : 1;
+	/** target state of the identifier */
+	union {
+		u32_t std_id   : 11;
+		u32_t ext_id   : 29;
+	};
+	/** rtr bit mask */
+	u32_t rtr_mask     : 1;
+	/** identifier mask*/
+	union {
+		u32_t std_id_mask  : 11;
+		u32_t ext_id_mask  : 29;
+	};
+} __packed;
+
+/**
+ * @typedef can_tx_callback_t
+ * @brief Define the application callback handler function signature
+ *
+ * @param error_flags status of the preformed send operation
+ */
+typedef void (*can_tx_callback_t)(u32_t error_flags);
+
+/**
+ * @typedef can_rx_callback_t
+ * @brief Define the application callback handler function signature
+ *        for receiving.
+ *
+ * @param received message
+ */
+typedef void (*can_rx_callback_t)(struct can_msg *msg);
+
+/**
+ * @brief Configure operation of a host controller.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param mode Operation mode
+ * @param bitrate bus-speed in Baud/s
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error, failed to configure device.
+ */
+typedef int (*can_configure_t)(struct device *dev, enum can_mode mode,
+							   u32_t bitrate);
```

Comment:
wrong parameter alignment: it should be aligned below previous '(' + 1 space.

Apply that comment everywhere, it's the same when calling functions etc...

Reply:
ohh. Now I got it. I was wondering why it always looked so bad after submitting. I used 4 spaces tabs instead of 8!